context.go

```diff
@@ -32,290 +65,45 @@ func (c *Context) FindVariable(e Expression) (Expression, error) {
-//     #      context['var'] = 'hi'
-//     #   end
-//     #
-//     #   context['var]  #=> nil
-//     def stack(new_scope = nil)
-//       old_stack_used = @this_stack_used
-//       if new_scope
-//         push(new_scope)
-//         @this_stack_used = true
-//       else
-//         @this_stack_used = false
-//       end
-
-//       yield
-//     ensure
-//       pop if @this_stack_used
-//       @this_stack_used = old_stack_used
-//     end
-
-//     def clear_instance_assigns
-//       @scopes[0] = {}
-//     end
-
-//     # Only allow String, Numeric, Hash, Array, Proc, Boolean or <tt>Liquid::Drop</tt>
-//     def []=(key, value)
-//       unless @this_stack_used
-//         @this_stack_used = true
-//         push({})
-//       end
-//       @scopes[0][key] = value
-//     end
-
-//     # Look up variable, either resolve directly after considering the name. We can directly handle
-//     # Strings, digits, floats and booleans (true,false).
-//     # If no match is made we lookup the variable in the current scope and
-//     # later move up to the parent blocks to see if we can resolve the variable somewhere up the tree.
-//     # Some special keywords return symbols. Those symbols are to be called on the rhs object in expressions
-//     #
-//     # Example:
-//     #   products == empty #=> products.empty?
-//     def [](expression)
-//       evaluate(Expression.parse(expression))
-//     end
-
-//     def key?(key)
-//       self[key] != nil
-//     end
-
-//     def evaluate(object)
-//       object.respond_to?(:evaluate) ? object.evaluate(self) : object
-//     end
-
-//     # Fetches an object starting at the local scope and then moving up the hierachy
-//     def find_variable(key)
-//       # This was changed from find() to find_index() because this is a very hot
-//       # path and find_index() is optimized in MRI to reduce object allocation
-//       index = @scopes.find_index { |s| s.key?(key) }
-//       scope = @scopes[index] if index
-
-//       variable = nil
-
-//       if scope.nil?
-//         @environments.each do |e|
-//           variable = lookup_and_evaluate(e, key)
-//           unless variable.nil?
-//             scope = e
-//             break
-//           end
-//         end
-//       end
-
-//       scope ||= @environments.last || @scopes.last
-//       variable ||= lookup_and_evaluate(scope, key)
-
-//       variable = variable.to_liquid
-//       variable.context = self if variable.respond_to?(:context=)
-
-//       variable
-//     end
-
-//     def lookup_and_evaluate(obj, key)
-//       if @strict_variables && obj.respond_to?(:key?) && !obj.key?(key)
-//         raise Liquid::UndefinedVariable, "undefined variable #{key}"
-//       end
+// Adds a new scope to the scopeStack
+func (s *scopeStack) push() {
+	*s = append(*s, Vars{})
+}
 
-//       value = obj[key]
+// Removes scope from the scopeStack
+func (s *scopeStack) pop() error {
+	l := len(*s)
+	if l < 1 {
+		return errors.New(`No scopes to pop`)
```

Comment:
tend to prefer factoring out into something like `ErrNoScopes = errors.New("No scopes to pop")` since it gets used in a few places